src/main/scala/wdl4s/types/WdlPairType.scala

```diff
@@ -19,6 +24,26 @@ case class WdlPairType(leftType: WdlType, rightType: WdlType) extends WdlType {
   override protected def coercion: PartialFunction[Any, WdlValue] = {
     case otherPair @ WdlPair(otherValue1, otherValue2) if isCoerceableFrom(otherPair.wdlType) =>
       WdlPair(leftType.coerceRawValue(otherValue1).get, rightType.coerceRawValue(otherValue2).get)
+    case jsObject: JsObject if jsObject.fields.size == 2 => mapToPair(jsObject.fields, this)
+  }
+
+  def mapToPair(m: Map[String, JsValue], wdlPairType: WdlPairType): WdlPair = {
+
+    val wdlPair: Seq[Try[WdlValue]] = (m.get("Left"), m.get("Right")) match {
+      case (Some(leftVal), Some(rightVal)) => Seq(wdlPairType.leftType.coerceRawValue(leftVal), wdlPairType.rightType.coerceRawValue(rightVal))
```

Comment:
Shouldn't there be another `case _ => ...` to account for when it's not a valid json ? (and associated test case)